src/ansys_sphinx_theme/theme/ansys_sphinx_theme/components/search-field.html

```diff
@@ -1,80 +1,149 @@
-  {% endif %}
-  <script>
-  var HOST_URL = "{{ HOST_URL }}";
-  var API_KEY = "{{ API_KEY }}";
-  // Iterates of the dictionary of index_uids and index_names to get the first child dicts
-  // and then the first key of the child dict to get the index_uid
-  var indexUid = "{{ theme_use_meilisearch.index_uids.items()|first|first }}";
-
-  // Function to check if the MeiliSearch host URL is reachable
-  function checkMeiliSearchHost(url, callback) {
-    var xhr = new XMLHttpRequest();
-    xhr.open('HEAD', url, true);
-    xhr.onload = function () {
-      if (xhr.status === 404) {
-        callback(false);
+
+      //if query is empty,  dont show the div result container
+      if (query === '') {
+        resultsContainer.style.display = 'none';
+        return;
       } else {
-        callback(true);
+        resultsContainer.style.display = 'block';
       }
-    };
-    xhr.onerror = function () {
-      callback(false);
-    };
-    xhr.send();
-  }
-
-  // Check the MeiliSearch host URL and load the appropriate search script
-  checkMeiliSearchHost(HOST_URL, function (isReachable) {
-    if (isReachable) {
-      var script = document.createElement('script');
-      script.src = "{{ pathto('_static/js/meilisearch_theme_wrap.js', 1) }}";
-      document.head.appendChild(script);
-      // create an env value that the url is reachable to use in another script
-      const ACTIVE_MEILI_BACKEND = true;
+
+      results.forEach(result => {
+        const item = document.createElement('div');
+        item.className = 'result-item';
+        const { title, text, href } = result.item;
+
+        const highlightedText = highlightTerms(text, query);
+        const highlightedTitle = highlightTerms(title, query);
+
+        item.innerHTML =
+          `<div class="result-title">${highlightedTitle}</div>` +
+          `<div class="result-text">${highlightedText}</div>`;
+        item.setAttribute('data-href', href); // Add href data attribute
+
+        // Add click event listener to navigate to the URL
+        item.addEventListener('click', () => {
+          navigateToHref(href);
+        });
+
+        resultsContainer.appendChild(item);
+      });
+    }
+
+    function stripText(text, query) {
+      // get only the sentences that contain the query
+      return text.replace(/<[^>]*>?/gm, '').split('.').filter((sentence) => sentence.includes(query)).join('. ');
     }
+
+    function highlightTerms(text, query) {
+    if (!query.trim()) return text;
+
+    // Split query into words and escape special characters for each word
+    const words = query.trim().split(/\s+/);
+    const escapedWords = words.map(word => word.replace(/[-\/\\^$*+?.()|[\]{}]/g, '\\$&'));
+
+    // Create a regex to match any of the words
+    const regex = new RegExp(`(${escapedWords.join('|')})`, 'gi');
+
+    // Replace matched text with highlighted version
+    return text.replace(regex, '<span class="highlight">$1</span>');
+}
```

Comment:
Adding some indentation:

```suggestion
    function highlightTerms(text, query) {
      if (!query.trim()) return text;

      // Split query into words and escape special characters for each word
      const words = query.trim().split(/\s+/);
      const escapedWords = words.map(word => word.replace(/[-\/\\^$*+?.()|[\]{}]/g, '\\$&'));

      // Create a regex to match any of the words
      const regex = new RegExp(`(${escapedWords.join('|')})`, 'gi');

      // Replace matched text with highlighted version
      return text.replace(regex, '<span class="highlight">$1</span>');
}
```